SCRUM-1203 Updates for DQM submission of DAs to new schema

markquintontulloch opened the pull request.

markquintontulloch committed:
Create and update DTOs for DA ingest

markquintontulloch committed:
Add bulk load executor code for DA DQM submissions

markquintontulloch committed:
Fix issue with underscore in loadType

markquintontulloch committed:
Update services for LinkML schema bulk DA submissions

markquintontulloch committed:
Update controllers and interfaces for LinkML schema bulk DA submissions

markquintontulloch committed:
Updates due to LinkML changes

markquintontulloch committed:
Update UI for singleReference change

markquintontulloch committed:
Correct entity type DAOs

oblodgett commented on src/main/cliapp/src/containers/diseaseAnnotationsPage/DiseaseAnnotationsTable.js:
```diff
@@ -445,11 +445,11 @@ export const DiseaseAnnotationsTable = () => {
       sortable: isEnabled,
       filter: true,
-      filterElement: filterComponentTemplate("reference", ["reference.curie"])
+      filterElement: filterComponentTemplate("singleReference", ["singleReference.curie"])
```
singleReference should be singleReferenceFilter just to differentiate from the field name.

markquintontulloch replied:
all filter names updated for clarity

oblodgett commented on src/main/java/org/alliancegenome/curation_api/interfaces/crud/AGMDiseaseAnnotationCrudInterface.java:
```diff
@@ -34,4 +37,37 @@
+    
+    @POST @Secured
+    @Path("/bulk/sgdAnnotationFile")
+    public String updateSgdAgmDiseaseAnnotations(List<DiseaseAnnotationDTO> annotationData);
```
@markquintontulloch why do we need to split these out into each their own? Considering the definition hasn't changed?

markquintontulloch replied:
Given that the ingest class splits disease annotations into the separate entity classes (i.e. each gets submitted in its own array in the ingest file), in my mind it makes more sense to be able to submit each entity separately.  In this way a MOD can resubmit, for example, just gene disease annotations without removing all agm and allele disease annotations.

oblodgett replied:
Ok... then to my later comments we might want to have the three different types also for `DiseaseAnnotationDTO`

markquintontulloch replied:
done

oblodgett commented on src/main/java/org/alliancegenome/curation_api/jobs/BulkLoadJobExecutor.java:
```diff
@@ -53,6 +54,18 @@
+            "WB", "NCBITaxon:6239",
+            "ZFIN", "NCBITaxon:7955"
+            );  
+
```
This should probably be split out into its own enum see:
https://github.com/alliance-genome/agr_java_software/blob/master/agr_java_core/src/main/java/org/alliancegenome/neo4j/entity/SpeciesType.java
for reference.

oblodgett replied:
I see both the list and the enum now?

markquintontulloch replied:
Yes, not sure how to deal with this one.  The map is still there for the FMS load.  The datatype for this is a string, not an enum.  I don't know if we want to change this.  Is there a reason for this being different to the backendbulkloadtype used for the manual load?

oblodgett commented on src/main/java/org/alliancegenome/curation_api/model/entities/PaperHandle.java:
```diff
@@ -21,7 +21,7 @@
     @ManyToOne
     @JsonView({View.FieldsOnly.class})
-    private Reference reference;
+    private Reference singleReference;
```
Did we discuss getting rid of this class? (Paperhandle that is)

markquintontulloch replied:
removed

oblodgett commented on src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java:
```diff
@@ -58,173 +78,93 @@ protected void init() {
+        } else if (entityType.equals("allele")) {
+            annotationsIdsBefore.addAll(alleleDiseaseAnnotationDAO.findAllAnnotationIds(taxonID));
+        } else if (entityType.equals("gene")) {
+            annotationsIdsBefore.addAll(geneDiseaseAnnotationDAO.findAllAnnotationIds(taxonID));
```
So... I think a better approach here would be to have the 3 different services... that extend DiseaseAnnotationService the common stuff put in the DiseaseAnnotationService and the specific stuff leave in each.

markquintontulloch replied:
agreed, have refactored

oblodgett commented on src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java:
```diff
@@ -250,59 +190,225 @@ public void runLoad(String taxonID, List<DiseaseAnnotationDTO> annotations) {
+                annotation = agmAnnotation;
+            } else {
+                annotation = annotationList.getResults().get(0);
+            }
```
Splitting if up would prevent these huge if else if's.

oblodgett commented on src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java:
```diff
@@ -250,59 +190,225 @@ public void runLoad(String taxonID, List<DiseaseAnnotationDTO> annotations) {
+        return annotation;
+    }
+    
+    private ExperimentalCondition validateExperimentalConditionDTO(ExperimentalConditionDTO dto) {
```
This whole method might be better served put in the ExperimentalConditionService class.

markquintontulloch replied:
done

oblodgett commented:
@markquintontulloch please don't hate me.

markquintontulloch committed:
Update filter names

markquintontulloch committed:
Remove PaperHandle class

markquintontulloch committed:
Create separate loads for AGM/allele/gene disease annotations

Retain DISEASE_ANNOTATION bulkLoadType so that all 3 can be loaded
with a single bulk load.

markquintontulloch committed:
Move validateExperimentalConditionDTO method to EC service

markquintontulloch committed:
Use enum to get taxon for manual bulk load

markquintontulloch committed:
Merge branch 'alpha' into SCRUM-1203

oblodgett commented on src/main/java/org/alliancegenome/curation_api/jobs/BulkLoadJobExecutor.java:
```diff
@@ -111,21 +127,54 @@ public void process(BulkLoadFile bulkLoadFile) throws Exception {
             bulkLoadFileDAO.merge(bulkLoadFile);
-            diseaseService.runLoad(map.get(fms.getDataSubType()), diseaseData);
+            diseaseFmsService.runLoad(modTaxons.get(fms.getDataSubType()), diseaseData);
+        } else if(bulkLoadFile.getBulkLoad().getBackendBulkLoadType() == BackendBulkLoadType.DISEASE_ANNOTATION) {
```
I guess I would just drop the Disease Annotation load ALL together.

markquintontulloch commented:
@oblodgett - the only slight gotcha with separate loads is that should the DQM want to put all disease annotation types in a single file (but load them separately), when the second entity type is loaded the md5sum check shows that the file exists and the load doesn't happen (and the file doesn't get associated with the load).  As such, I've kept the DISEASE_ANNOTATION BackendBulkLoadType in addition to the entity-specific ones.  This allows all 3 entity types to be loaded in a single load, which is what I imagine most people would want anyway if they have put all the entities in a single file.

oblodgett commented on src/main/java/org/alliancegenome/curation_api/services/AlleleDiseaseAnnotationCrudService.java:
```diff
@@ -37,4 +54,77 @@ protected void init() {
+        log.debug("runLoad: Before: " + taxonId + " " + annotationIdsBefore.size());
+        List<String> annotationIdsAfter = new ArrayList<>();
+        ProcessDisplayHelper ph = new ProcessDisplayHelper(10000);
+        ph.startProcess("Disease Annotation Update " + taxonId, annotations.size());
```
Might want to change this to display the type that is running (AGM) and not just "Disease Annotation"

markquintontulloch replied:
Done

oblodgett commented:
But that could be true for adding in Gene's and Alleles and everything else that would fit in a "set"... So what I think might be better is to change DISEASE_ANNOATION -> FULL_INGEST then we will just go through ALL the sets and do the appropriate load for each one.

markquintontulloch committed:
Add FULL_INGEST bulk load type

markquintontulloch committed:
Add annotation type to load start log message

markquintontulloch commented:
> But that could be true for adding in Gene's and Alleles and everything else that would fit in a "set"... So what I think might be better is to change DISEASE_ANNOTATION -> FULL_INGEST then we will just go through ALL the sets and do the appropriate load for each one.

I've added FULL_INGEST but also kept DISEASE_ANNOTATION.  The reason being that DQMs are used to submitting all disease annotations as a single file, but other entities separately.  This would work with the WB workflow, for example, where it would be easy to generate all disease annotations together, but not so easy to generate all submissions in a single file.

oblodgett commented on src/main/java/org/alliancegenome/curation_api/model/entities/bulkloads/BulkLoad.java:
```diff
@@ -74,15 +75,32 @@
         ;
     }
     
     public enum BackendBulkDataType {
```
I think pulling this out into its own file for the enum.

oblodgett committed:
Refactored BackendBulkDataType enum

oblodgett committed:
Refactored loader

markquintontulloch committed:
Merge pull request #284 from alliance-genome/SCRUM-1203_refactor

Refactored loader

markquintontulloch committed:
Merge branch 'alpha' into SCRUM-1203
